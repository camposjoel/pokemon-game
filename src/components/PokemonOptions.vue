<template>
  <div class="options-container">
    <ul>
      <li v-for="pokemon in pokemons"
        :key="pokemon.id"
        @click="$emit('selection', pokemon.id)">
        {{ pokemon.name }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    pokemons: {
      type: Array,
      required: true
    }
  }
}
</script>

<style scoped>
ul {
  list-style-type: none;
  display: flex;
}
li {
  background-color: lightsalmon;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin: 15px 10px 0;
  padding: 10px 0;
  width: 250px;
  height: 32px;
  display: flex;
  justify-content: center;
  align-items: center;
  text-transform: capitalize;
  font-weight: 700;
}

li:hover {
  box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
}

.options-container {
  display: flex;
  justify-content: center;
}
</style>
