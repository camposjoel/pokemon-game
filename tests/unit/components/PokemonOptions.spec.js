// @vitest-environment happy-dom

import { describe, expect, test, beforeEach, should } from "vitest"
import { shallowMount } from '@vue/test-utils'
import PokemonOptions from '@/components/PokemonOptions.vue'
import { pokemons } from '../mocks/pokemons.mock'

describe('PokemonOptions Component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons
      }
    })
  })

  test('Should match with snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Should show 4 pokemons', () => {
    const pokemonList = wrapper.findAll('li')
    expect(pokemonList.length).toBe(4)
    expect(pokemonList[0].text()).toBe('bulbasaur')
    expect(pokemonList[1].text()).toBe('ivysaur')
    expect(pokemonList[2].text()).toBe('venusaur')
    expect(pokemonList[3].text()).toBe('charmander')
  })

  test('Should emit -selection- with parameters', () => {
    const [ p1, p2, p3, p4 ] = wrapper.findAll('li')

    p1.trigger('click')
    p2.trigger('click')
    p3.trigger('click')
    p4.trigger('click')
    expect(wrapper.emitted('selection').length).toBe(4)
    expect(wrapper.emitted('selection')[0]).toEqual([1])
    expect(wrapper.emitted('selection')[1]).toEqual([2])
    expect(wrapper.emitted('selection')[2]).toEqual([3])
    expect(wrapper.emitted('selection')[3]).toEqual([4])
  })

})