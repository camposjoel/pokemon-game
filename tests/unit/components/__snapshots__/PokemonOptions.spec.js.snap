// Vitest Snapshot v1

exports[`PokemonOptions Component > Should match with snapshot 1`] = `
"<div class=\\"options-container\\" data-v-53abcefb=\\"\\">
  <ul data-v-53abcefb=\\"\\">
    <li data-v-53abcefb=\\"\\">bulbasaur</li>
    <li data-v-53abcefb=\\"\\">ivysaur</li>
    <li data-v-53abcefb=\\"\\">venusaur</li>
    <li data-v-53abcefb=\\"\\">charmander</li>
  </ul>
</div>"
`;
